polars/polars-lazy/src/dsl/dt.rs

```diff
@@ -165,6 +168,23 @@ impl DateLikeNameSpace {
             .map(function, GetOutput::from_type(DataType::UInt32))
             .with_fmt("dt.second")
     }
+
+    /// Get the millisecond of a Datetime/Time64
+    pub fn millisecond(self) -> Expr {
```

Comment:
All `map` function should be replaced with `map_private`. This is a work in progress, but all new functions should be using that architecture to finish that WIP someday. :)

The benefit will be that the optimizer understands what happens and will not block certain optimizations and it will make the logical plan serializable.

Reply:
> ... all new functions should be using that architecture to finish that WIP someday. :)

Heh; painting the Golden Gate Bridge...
I'll take a look, see what the differences are, and try to get it done! 👍

Reply:
@ritchie46; not many temporal functions using the new architecture (think I saw _one_?), whereas the string functions look well underway.Want me to set-up a new namespace? (equivalent to `StringFunction`) .

eg: something like `pub enum TemporalFunction { ...`

Reply:
Yeap, that would be great 👍

Reply:
Ok, I'm going to split-out the `strptime` & misc fixes into a separate PR, independent of the addition of `dt.millisecond` and `dt.microsecond`. Then the fixes can get in sooner, and I can go on to look at reworking the two new methods without tying that to the fixes (and maybe bring along a few more of the temporal funcs while I'm at it ;)

Reply:
@ritchie46; ok, done - created a separate branch to work on the microsecond/millisecond update, leaving this one with just the `strptime` enhancements and the `fractional` second option.

Reply:
Alright!